index.d.ts

```diff
@@ -0,0 +1,23 @@
+declare namespace CanCan {
+    export interface Option {
+        instanceOf?: <T, U>(instance: T, model: U) => boolean;
```

Comment:
I'm not familiar with TS, but where are those T, U declared? Can they be explanatory?

Reply:
They are called generics, which is a technique to due with multiple types.

For example,
```
// this function accepts an arg and it must be an number.
// what about if I also want the function to accept string, boolean, etc?
function identity(arg: number): number {
    return arg;
}
```

```
// I could declare the type to any, which means it accepts any types. 
// however we are losing type information when using any.
function identity(arg: any): any {
    return arg;
}
```

```
// actually, we can use generic type to capture the type information
function identity<T>(arg: T): T {
    return arg;
}
```

A long explanation can be found in typescript handbook
https://www.typescriptlang.org/docs/handbook/generics.html